### PR TITLE
Refactor command line parsing + add trigger support

### PR DESCRIPTION
**Pull request by michaelcfine**
Added support for pre- and post- triggers
As part of doing this, refactored command line argument parsing into its own module

### CODE REVIEW

**Comment by michaelcfine** [comment]
👍 but the links on the three buttons/images need to be corrected.

File: server/src/triggers/lifecycle.close/post.mjs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+* SPDX-License-Identifier: Apache-2.0
+*/
+
+function post(ctx, params) {
+  ctx.setTimeout(function() {
+    const result = { state: 'inactive', previous: ctx.fireboltCoreSdk.Lifecycle.state() };
```
Comment (jlacivita) [inline server/src/triggers/lifecycle.close/post.mjs:21]: FYI, the Core SDK manages the value of `previous` so you can leave it out if you want.

**Review by jlacivita — DISMISSED** [review]
LGTM

File: server/src/triggers/lifecycle.close/post.mjs (hunk 1 above)
Comment (michaelcfine) [reply server/src/triggers/lifecycle.close/post.mjs]: Good call. Have removed 'previous' from both post.mjs files.

File: server/src/commandLine.mjs [hunk 2]
```diff
@@ -0,0 +1,71 @@
+/*
```
[the remark this answers is not in the document]
Comment (michaelcfine) [reply server/src/commandLine.mjs:1]: This isn't the top-level file you run, though. Okay to leave as-is?

**Review by jlacivita — APPROVED** [review]

**Pull request merged**